lib/linguist/heuristics.rb

```diff
@@ -420,6 +420,14 @@ def call(data)
       end
     end
 
+    disambiguate ".sol" do |data|
+      if /^pragma solidity/.match(data)
+        Language["Solidity"]
+      elsif /M02\*$/.match(data)
```

Comment:
`M02` is a [Gerber](https://www.ucamco.com/files/downloads/file/81/the_gerber_file_format_specification.pdf) command code, not Eagle data. Specifically, an EOF marker that the spec mandates must terminate every Gerber image file. See #3744 for details.

(It would also behove you to check out #3743 for a total rehaul of the site's current definition of KiCad/Eagle PCB data. Eagle isn't so much affected, but KiCad is; and it turned out to be three distinct formats instead of one...)